.github/workflows/tests-e2e-upgrade-restart.yaml

```diff
@@ -0,0 +1,816 @@
+name: Cilium E2E Upgrade Restart (ci-e2e-upgrade-restart)
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  workflow_dispatch:
+    inputs:
+      PR-number:
+        description: "Pull request number."
+        required: true
+      context-ref:
+        description: "Context in which the workflow runs. If PR is from a fork, will be the PR target branch (general case). If PR is NOT from a fork, will be the PR branch itself (this allows committers to test changes to workflows directly from PRs)."
+        required: true
+      SHA:
+        description: "SHA under test (head of the PR branch)."
+        required: true
+      extra-args:
+        description: "[JSON object] Arbitrary arguments passed from the trigger comment via regex capture group. Parse with 'fromJson(inputs.extra-args).argName' in workflow."
+        required: false
+        default: '{}'
+  push:
+    branches:
+      - 'renovate/main-**'
+  # Run every 8 hours
+  schedule:
+    - cron:  '0 5/8 * * *'
+
```

Comment:
Heads up that in order to test this, the best way for you to iterate would be to open this PR on your own GitHub fork and add `pull_request: {}` into the triggers so that the workflow can run directly on your fork.

As-is, this workflow will not be run as part of the PR, because we have to be a bit careful around triggering arbitrary workflows from contributors for supply chain security reasons.

When you're happy with the changes, a reviewer or committer could copy the PR into a branch on the Cilium organization and trigger the test with a fresh PR (also with the added trigger for `pull_request: {}`).

Reply:
Thanks for the tip. I'll play around with it and see if I can get the workflow to run in my fork first.

> When you're happy with the changes, a reviewer or committer could copy the PR into a branch on the Cilium organization and trigger the test with a fresh PR (also with the added trigger for pull_request: {}).

@hemanthmalla helped me last time with this, so maybe I'll see if he could help with this one.